Documentation/cmdref/cilium-agent.md

```diff
@@ -167,7 +167,6 @@ cilium-agent [flags]
       --encryption-strict-mode-cidr string                        In strict-mode encryption, all unencrypted traffic coming from this CIDR and going to this same CIDR will be dropped
       --endpoint-bpf-prog-watchdog-interval duration              Interval to trigger endpoint BPF programs load check watchdog (default 30s)
       --endpoint-queue-size int                                   Size of EventQueue per-endpoint (default 25)
-      --endpoint-status strings                                   Enable additional CiliumEndpoint status features (controllers,health,log,policy,state)
```

Comment:
I see https://github.com/cilium/cilium/issues/29804 is specific to the `policy` option. Have you performed similar tests with the other options?

Reply:
I mostly did educated guess, I only had data for `policy` from the production cluster that was bloating Etcd data to 8GB limit with only 1k pods because of multiple revisions of the same CEP. Essentially breaking the whole Kubernetes control plane when Etcd data limit was reached. Not counting unnecessary CPU burnt on processing CEP endpoint events in agent etc

For other options:
- controllers is reporting consecutive-failures and failure count - it would cause exactly the same issue
- log - same as above, pushing log updates of controllers would cause multiple updates as well and unnecessary churn
- state is still being reported - this option had no effect before and it's still present in the status
- health - it's almost the same as `state`, only makes additional check on logs of controllers, not that useful, can be replaced with metrics.

For the deprecation policy, I was thinking about deprecating that first, but taking into account how harmful these options are and potentially only used for debugging, not production environments, I decided to remove it without the deprecation first. One less option to shoot yourself in the foot.